### PR TITLE
MOTECH-2385: Discards openmrs19.resource.model package

### PR DESCRIPTION
**Pull request by kmadej**
This PR discards resource.model package in OpenMRS Module. It required big
refactor in OpenMRS.

### CODE REVIEW

**Comment by kmadej** [comment]
test this please

File: openmrs-19/src/main/java/org/motechproject/openmrs19/domain/Encounter.java [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-import java.lang.reflect.Type;
 import java.util.Date;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * Represents a single OpenMRS encounter. An encounter is a single, specific interaction between the patient and a
  * provider. It can be any interaction and includes doctor visits, laboratory tests, food distribution, home visits,
- * counselor appointments, etc. It's a part of the OpenMRS model.
+ * counselor appointments, etc. It's a part of the MOTECH model.
```
Comment (pgesek) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/domain/Encounter.java]: These "it's part of" sentences can just go away now

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSEncounterServiceImpl.java [hunk 2]
```diff
@@ -327,18 +146,53 @@ public void deleteEncounterType(String uuid) {
...
+                observation.getConcept().setUuid(conceptUuid);
+                updatedObs.add(observation);
             }
         }
+
+        return updatedObs;
+    }
+
+    private List<Encounter> getAllEncountersByPatientMotechId(String motechId) {
+        Validate.notEmpty(motechId, "MoTeCH Id cannot be empty");
```
Comment (pgesek) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSEncounterServiceImpl.java]: bad capitalization of MOTECH

File: openmrs-19/src/test/java/org/motechproject/openmrs19/it/MRSFacilityServiceIT.java [hunk 3]
```diff
@@ -63,11 +63,11 @@ public void shouldCreateFacility() throws InterruptedException {
...
         assertFalse(mrsListener.updated);
         assertFalse(mrsListener.deleted);
-        assertEquals(facilityTwo.getName(), mrsListener.eventParameters.get(EventKeys.FACILITY_NAME));
-        assertEquals(facilityTwo.getCountry(), mrsListener.eventParameters.get(EventKeys.FACILITY_COUNTRY));
-        assertEquals(facilityTwo.getRegion(), mrsListener.eventParameters.get(EventKeys.FACILITY_REGION));
-        assertEquals(facilityTwo.getCountyDistrict(), mrsListener.eventParameters.get(EventKeys.FACILITY_COUNTY_DISTRICT));
-        assertEquals(facilityTwo.getStateProvince(), mrsListener.eventParameters.get(EventKeys.FACILITY_STATE_PROVINCE));
+        assertEquals(facilityTwo.getName(), mrsListener.eventParameters.get(EventKeys.LOCATION_NAME));
+        assertEquals(facilityTwo.getCountry(), mrsListener.eventParameters.get(EventKeys.LOCATION_COUNTRY));
+        assertEquals(facilityTwo.getAddress6(), mrsListener.eventParameters.get(EventKeys.LOCATION_REGION));
```
Comment (pgesek) [inline openmrs-19/src/test/java/org/motechproject/openmrs19/it/MRSFacilityServiceIT.java]: can we add a method ignored by json for this?

**Comment by kmadej** [comment]
I checked locally openMRS IT and it passed

**Comment by PiotrZurawskiSolDevelo** [comment]
Checked. Data source for Encounter is not display information about Patient. Bubble is named Encounter#0.Patient.display.

**Comment by kmadej** [comment]
@PiotrZurawskiSolDevelo I fixed it

**Comment by kmadej** [comment]
test this please

**Comment by PiotrZurawskiSolDevelo** [comment]
Checked. I Found one more little error in Encounter data source. I get as Type bubble this value: "org.motechproject.openmrs19.domain.EncounterType@2d629756".

**Comment by kmadej** [comment]
@PiotrZurawskiSolDevelo I fixed it, now the name of the encounterType is returned.

**Comment by PiotrZurawskiSolDevelo** [comment]
Checked. @kmadej everything is working now.

**Pull request merged**